opte-api/src/ip.rs

```diff
@@ -237,6 +237,39 @@ impl TryFrom<u8> for Protocol {
     }
 }
 
+impl TryFrom<smoltcp::wire::IpProtocol> for Protocol {
```

Comment:
So unfortunately this would require that clients (like sled-agent) pull in smoltcp, which isn't great. We really want this to remain an implementation detail of opte itself.

Reply:
Makes sense. I'll remove it, and put some conversions into the `ip6` module directly. We can figure out how to make them more accessible if we need to later.

Reply:
Actually, I just realized that I didn't have to update `Cargo.toml`, because it's already in there. Is that intentional? I'll just move these conversions now, we can strip that out later if a follow-up PR.

Reply:
Sigh, yea because I already did this myself. And I think I had to do that to satisfy the orphan rule, e.g. I couldn't implement `From<smoltcp::wire::Ipv4Address> for Ipv4Addr` in opte because the types are defined in opte-api now.